packages/camera/camera_android_camerax/test/camera2_camera_info_test.mocks.dart

```diff
@@ -0,0 +1,179 @@
+// Mocks generated by Mockito 5.4.4 from annotations
+// in camera_android_camerax/test/camera2_camera_info_test.dart.
+// Do not manually edit this file.
+
+// ignore_for_file: no_leading_underscores_for_library_prefixes
+import 'dart:async' as _i7;
+
+import 'package:camera_android_camerax/src/camera_info.dart' as _i6;
+import 'package:camera_android_camerax/src/camera_state.dart' as _i8;
+import 'package:camera_android_camerax/src/exposure_state.dart' as _i3;
+import 'package:camera_android_camerax/src/live_data.dart' as _i2;
+import 'package:camera_android_camerax/src/zoom_state.dart' as _i9;
+import 'package:mockito/mockito.dart' as _i1;
+import 'package:mockito/src/dummies.dart' as _i5;
+
+import 'test_camerax_library.g.dart' as _i4;
+
+// ignore_for_file: type=lint
+// ignore_for_file: avoid_redundant_argument_values
+// ignore_for_file: avoid_setters_without_getters
+// ignore_for_file: comment_references
+// ignore_for_file: deprecated_member_use
+// ignore_for_file: deprecated_member_use_from_same_package
+// ignore_for_file: implementation_imports
+// ignore_for_file: invalid_use_of_visible_for_testing_member
+// ignore_for_file: prefer_const_constructors
+// ignore_for_file: unnecessary_parenthesis
+// ignore_for_file: camel_case_types
+// ignore_for_file: subtype_of_sealed_class
+
+class _FakeLiveData_0<T extends Object> extends _i1.SmartFake
+    implements _i2.LiveData<T> {
+  _FakeLiveData_0(
+    Object parent,
+    Invocation parentInvocation,
+  ) : super(
+          parent,
+          parentInvocation,
+        );
+}
+
+class _FakeExposureState_1 extends _i1.SmartFake implements _i3.ExposureState {
+  _FakeExposureState_1(
+    Object parent,
+    Invocation parentInvocation,
+  ) : super(
+          parent,
+          parentInvocation,
+        );
+}
+
+/// A class which mocks [TestCamera2CameraInfoHostApi].
+///
+/// See the documentation for Mockito's code generation for more information.
+class MockTestCamera2CameraInfoHostApi extends _i1.Mock
+    implements _i4.TestCamera2CameraInfoHostApi {
+  MockTestCamera2CameraInfoHostApi() {
+    _i1.throwOnMissingStub(this);
+  }
+
+  @override
+  int createFrom(int? cameraInfoIdentifier) => (super.noSuchMethod(
+        Invocation.method(
+          #createFrom,
+          [cameraInfoIdentifier],
+        ),
+        returnValue: 0,
+      ) as int);
+
+  @override
+  int getSupportedHardwareLevel(int? identifier) => (super.noSuchMethod(
+        Invocation.method(
+          #getSupportedHardwareLevel,
+          [identifier],
+        ),
+        returnValue: 0,
```

Comment:
Consider a comment saying this is LIMITED. It coudl help in debugging later to know that the device might not have returned limited but we had an error. https://developer.android.com/reference/android/hardware/camera2/CameraMetadata#INFO_SUPPORTED_HARDWARE_LEVEL_LIMITED

Reply:
This is generated code :/ I think it's just a mock value